### PR TITLE
Bookstore: Connect to API

### PR DESCRIPTION
**Pull request by Ghiftee**
In this milestone, I refactored the redux code of my application using middleware to make async requests to the API.

### CODE REVIEW

File: src/api/bookstore.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    body: JSON.stringify({
+      item_id: book.id,
+      title: `${book.title}&|&|&${book.author}`,
+      category: book.category,
+    }),
+  });
+  const status = await response.text();
+  return status;
+};
+
```
Comment (dasileker) [inline src/api/bookstore.js:36]: - [x] The `**AddNewBook**` is  not stored in the redux store, and also I can't tell which is the title and the author see the screenshot 👇


nnn

File: src/api/bookstore.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export const initialize = async () => {
+  if (!localStorage.getItem('appId')) {
+    const response = await fetch(`${BASE_URL}/apps/`, {
+      method: 'POST',
+    });
+    const appId = await response.text();
+    localStorage.setItem('appId', appId);
+  }
+};
```
Comment (dasileker) [inline src/api/bookstore.js:11]: - [x] And also the fetch method for the API is not rendering anything  👉 [see the screenshot here](https://user-images.githubusercontent.com/61562626/130690319-17f1d8f1-9abc-42e5-bfa4-35f1da143441.png).

- [x] Please use the `componentDidMount` lifecycle method or with react hooks requests. see the screenshot 👇

nbnbn

File: src/api/bookstore.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: book.id,
+      title: `${book.title}&|&|&${book.author}`,
+      category: book.category,
+    }),
+  });
+  const status = await response.text();
+  return status;
```
Comment (dasileker) [inline src/api/bookstore.js:34]: - [x] The `localstorage` is set but it's not storing anything. see screenshot 👇

bnbjkbjlv

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Ghiftee** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/api/bookstore.js (hunk 1 above)
Comment (Ghiftee) [reply src/api/bookstore.js:36]: Thank you for the review. However, I would like to clarify a few things:

- On line 46 of redux/books/book.js, I added the book to the redux store.
- The reason you can't tell which is the title and author is that there is no provision for the author to be stored on the server we were given so I added it to the title instead.

File: src/api/bookstore.js (hunk 2 above)
Comment (Emmyn5600) [reply src/api/bookstore.js:11]: Well done to use hooks.

File: src/api/bookstore.js (hunk 3 above)
Comment (Emmyn5600) [reply src/api/bookstore.js:34]: Well done here.

react4

**Review by Emmyn5600 — APPROVED** [review]
Hi, 

Good job so far! 👏🏼 @Ghiftee 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous TSE
 
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**